File: fixtures/coding/maintainability-mini-service-python/service/container.py
"""Application setup and dependency wiring."""

from __future__ import annotations

from service.audit import AuditLog
from service.handlers.customers import bulk_import_customers, create_customer, list_customers
from service.repository import CustomerRepository
from service.router import Application


def build_app() -> Application:
    """Build the mini application and register routes."""
    app = Application(repository=CustomerRepository(), audit_log=AuditLog())
    app.add_route("GET", "/customers", list_customers)
    app.add_route("POST", "/customers", create_customer)
    app.add_route("POST", "/customers/bulk-import", bulk_import_customers)
    return app

File: fixtures/coding/maintainability-mini-service-ts/src/container.ts
// Application setup and dependency wiring.

import { AuditLog } from "./audit";
import { bulkImportCustomers, createCustomer, listCustomers } from "./handlers/customers";
import { CustomerRepository } from "./repository";
import { Application } from "./router";

export function buildApp(): Application {
  const app = new Application(new CustomerRepository(), new AuditLog());
  app.addRoute("GET", "/customers", listCustomers);
  app.addRoute("POST", "/customers", createCustomer);
  app.addRoute("POST", "/customers/bulk-import", bulkImportCustomers);
  return app;
}
